provisioner/aws-pricing.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+var Promise = require('promise');
+var lodash = require('lodash');
+var util = require('util');
+var Cache = require('../cache');
+var assert = require('assert');
+
+
+function fetchPricing(ec2) {
+  var that = this;
+  var startDate = new Date();
+  startDate.setHours(startDate.getHours() - 2);
```

Comment:
Hmm... Couldn't we cache something like 3 hours of data-points in memory and then just fetch from the last data-point we have...